### ISSUE TITLE
Raw object data: extract txt stream with actual panning metadata

### ISSUE THREAD

**Issue by ValZapod** [VoidXH/Cavern#5]
That is what Huffman tables extracts, right? Raw object data is not enough! Since some of those objects are dynamic and so require txt file with position metadata, that works like subtitles.

**Comment by VoidXH** [comment]
Raw object data was basically a hack in the export process, I've added a better option in e9f1bcb644e8eb911be75984fe90661f2fec3753: for ADM BWF exports, a separation can be enabled by selecting XML as export format.

**Comment by ValZapod** [comment]
YES! Thank you, of course the per ETSI way is better!

**Comment by ValZapod** [comment]
Mediainfo now decodes:


```
Audio
Format                      : PCM
Format settings             : Little / Signed
Codec ID                    : 1
Duration                    : 3 min 7 s
Bit rate mode               : Constant
Bit rate                    : 18.4 Mb/s
Channel(s)                  : 16 channels
Sampling rate               : 48.0 kHz
Bit depth                   : 24 bits
Stream size                 : 412 MiB (96%)
Metadata format             : ADM, Version 0
Metadata muxing mode        : axml
Number of programmes        : 1
Number of contents          : 2
Number of objects           : 16
Number of pack formats      : 16
Number of channel formats   : 16
Number of track UIDs        : 16
Number of track formats     : 16
Number of stream formats    : 16
Programme #1                : Cavern_Export
 ID                         : APR_1001
 Title                      : Cavern_Export
 Start                      : 00:00:00.00000
 End                        : 00:03:07.58400
 Content #s                 : 1 + 2
Content #1                  : Channels
 ID                         : ACO_1001
 Title                      : Channels
 Mode                       : Mixed
 Object #s                  : 1
Content #2                  : Objects
 ID                         : ACO_1002
 Title                      : Objects
 Mode                       : Mixed
 Object #s                  : 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 + 11 + 12 + 13 + 14 + 15 + 16
Object #1                   : RoomCentricCenter
 ID                         : AO_1001
 Title                      : RoomCentricCenter
 Duration                   : 00:03:07.58400
 Pack format #s             : 1
 Track UID #s               : 1
Object #2                   : Audio Object 1
 ID                         : AO_1002
 Title                      : Audio Object 1
 Duration                   : 00:03:07.58400
 Pack format #s             : 2
 Track UID #s               : 2
Object #3                   : Audio Object 2
 ID                         : AO_1003
 Title                      : Audio Object 2
 Duration                   : 00:03:07.58400
 Pack format #s             : 3
 Track UID #s               : 3
Object #4                   : Audio Object 3
 ID                         : AO_1004
 Title                      : Audio Object 3
 Duration                   : 00:03:07.58400
 Pack format #s             : 4
 Track UID #s               : 4
Object #5                   : Audio Object 4
 ID                         : AO_1005
 Title                      : Audio Object 4
 Duration                   : 00:03:07.58400
 Pack format #s             : 5
 Track UID #s               : 5
Object #6                   : Audio Object 5
 ID                         : AO_1006
 Title                      : Audio Object 5
 Duration                   : 00:03:07.58400
 Pack format #s             : 6
 Track UID #s               : 6
Object #7                   : Audio Object 6
 ID                         : AO_1007
 Title                      : Audio Object 6
 Duration                   : 00:03:07.58400
 Pack format #s             : 7
 Track UID #s               : 7
Object #8                   : Audio Object 7
 ID                         : AO_1008
 Title                      : Audio Object 7
 Duration                   : 00:03:07.58400
 Pack format #s             : 8
 Track UID #s               : 8
Object #9                   : Audio Object 8
 ID                         : AO_1009
 Title                      : Audio Object 8
 Duration                   : 00:03:07.58400
 Pack format #s             : 9
 Track UID #s               : 9
Object #10                  : Audio Object 9
 ID                         : AO_100a
 Title                      : Audio Object 9
 Duration                   : 00:03:07.58400
 Pack format #s             : 10
 Track UID #s               : 10
Object #11                  : Audio Object 10
 ID                         : AO_100b
 Title                      : Audio Object 10
 Duration                   : 00:03:07.58400
 Pack format #s             : 11
 Track UID #s               : 11
Object #12                  : Audio Object 11
 ID                         : AO_100c
 Title                      : Audio Object 11
 Duration                   : 00:03:07.58400
 Pack format #s             : 12
 Track UID #s               : 12
Object #13                  : Audio Object 12
 ID                         : AO_100d
 Title                      : Audio Object 12
 Duration                   : 00:03:07.58400
 Pack format #s             : 13
 Track UID #s               : 13
Object #14                  : Audio Object 13
 ID                         : AO_100e
 Title                      : Audio Object 13
 Duration                   : 00:03:07.58400
 Pack format #s             : 14
 Track UID #s               : 14
Object #15                  : Audio Object 14
 ID                         : AO_100f
 Title                      : Audio Object 14
 Duration                   : 00:03:07.58400
 Pack format #s             : 15
 Track UID #s               : 15
Object #16                  : Audio Object 15
 ID                         : AO_1010
 Title                      : Audio Object 15
 Duration                   : 00:03:07.58400
 Pack format #s             : 16
 Track UID #s               : 16
PackFormat #1               : RoomCentricCenter
 ID                         : AP_00011001
 Title                      : RoomCentricCenter
 TypeDefinition             : DirectSpeakers
 Channel layout             : C
 Channel format #s          : 1
PackFormat #2               : Cavern_Obj_1
 ID                         : AP_00031001
 Title                      : Cavern_Obj_1
 TypeDefinition             : Objects
 Channel format #s          : 2
PackFormat #3               : Cavern_Obj_2
 ID                         : AP_00031002
 Title                      : Cavern_Obj_2
 TypeDefinition             : Objects
 Channel format #s          : 3
PackFormat #4               : Cavern_Obj_3
 ID                         : AP_00031003
 Title                      : Cavern_Obj_3
 TypeDefinition             : Objects
 Channel format #s          : 4
PackFormat #5               : Cavern_Obj_4
 ID                         : AP_00031004
 Title                      : Cavern_Obj_4
 TypeDefinition             : Objects
 Channel format #s          : 5
PackFormat #6               : Cavern_Obj_5
 ID                         : AP_00031005
 Title                      : Cavern_Obj_5
 TypeDefinition             : Objects
 Channel format #s          : 6
PackFormat #7               : Cavern_Obj_6
 ID                         : AP_00031006
 Title                      : Cavern_Obj_6
 TypeDefinition             : Objects
 Channel format #s          : 7
PackFormat #8               : Cavern_Obj_7
 ID                         : AP_00031007
 Title                      : Cavern_Obj_7
 TypeDefinition             : Objects
 Channel format #s          : 8
PackFormat #9               : Cavern_Obj_8
 ID                         : AP_00031008
 Title                      : Cavern_Obj_8
 TypeDefinition             : Objects
 Channel format #s          : 9
PackFormat #10              : Cavern_Obj_9
 ID                         : AP_00031009
 Title                      : Cavern_Obj_9
 TypeDefinition             : Objects
 Channel format #s          : 10
PackFormat #11              : Cavern_Obj_10
 ID                         : AP_0003100a
 Title                      : Cavern_Obj_10
 TypeDefinition             : Objects
 Channel format #s          : 11
PackFormat #12              : Cavern_Obj_11
 ID                         : AP_0003100b
 Title                      : Cavern_Obj_11
 TypeDefinition             : Objects
 Channel format #s          : 12
PackFormat #13              : Cavern_Obj_12
 ID                         : AP_0003100c
 Title                      : Cavern_Obj_12
 TypeDefinition             : Objects
 Channel format #s          : 13
PackFormat #14              : Cavern_Obj_13
 ID                         : AP_0003100d
 Title                      : Cavern_Obj_13
 TypeDefinition             : Objects
 Channel format #s          : 14
PackFormat #15              : Cavern_Obj_14
 ID                         : AP_0003100e
 Title                      : Cavern_Obj_14
 TypeDefinition             : Objects
 Channel format #s          : 15
PackFormat #16              : Cavern_Obj_15
 ID                         : AP_0003100f
 Title                      : Cavern_Obj_15
 TypeDefinition             : Objects
 Channel format #s          : 16
ChannelFormat #1            : RoomCentricCenter
 ID                         : AC_00011001
 Title                      : RoomCentricCenter
 TypeDefinition             : DirectSpeakers
 Channel layout             : C
 Type                       : Static
ChannelFormat #2            : Cavern_Obj_1
 ID                         : AC_00031001
 Title                      : Cavern_Obj_1
 TypeDefinition             : Objects
ChannelFormat #3            : Cavern_Obj_2
 ID                         : AC_00031002
 Title                      : Cavern_Obj_2
 TypeDefinition             : Objects
ChannelFormat #4            : Cavern_Obj_3
 ID                         : AC_00031003
 Title                      : Cavern_Obj_3
 TypeDefinition             : Objects
ChannelFormat #5            : Cavern_Obj_4
 ID                         : AC_00031004
 Title                      : Cavern_Obj_4
 TypeDefinition             : Objects
ChannelFormat #6            : Cavern_Obj_5
 ID                         : AC_00031005
 Title                      : Cavern_Obj_5
 TypeDefinition             : Objects
ChannelFormat #7            : Cavern_Obj_6
 ID                         : AC_00031006
 Title                      : Cavern_Obj_6
 TypeDefinition             : Objects
ChannelFormat #8            : Cavern_Obj_7
 ID                         : AC_00031007
 Title                      : Cavern_Obj_7
 TypeDefinition             : Objects
ChannelFormat #9            : Cavern_Obj_8
 ID                         : AC_00031008
 Title                      : Cavern_Obj_8
 TypeDefinition             : Objects
ChannelFormat #10           : Cavern_Obj_9
 ID                         : AC_00031009
 Title                      : Cavern_Obj_9
 TypeDefinition             : Objects
ChannelFormat #11           : Cavern_Obj_10
 ID                         : AC_0003100a
 Title                      : Cavern_Obj_10
 TypeDefinition             : Objects
ChannelFormat #12           : Cavern_Obj_11
 ID                         : AC_0003100b
 Title                      : Cavern_Obj_11
 TypeDefinition             : Objects
ChannelFormat #13           : Cavern_Obj_12
 ID                         : AC_0003100c
 Title                      : Cavern_Obj_12
 TypeDefinition             : Objects
ChannelFormat #14           : Cavern_Obj_13
 ID                         : AC_0003100d
 Title                      : Cavern_Obj_13
 TypeDefinition             : Objects
ChannelFormat #15           : Cavern_Obj_14
 ID                         : AC_0003100e
 Title                      : Cavern_Obj_14
 TypeDefinition             : Objects
ChannelFormat #16           : Cavern_Obj_15
 ID                         : AC_0003100f
 Title                      : Cavern_Obj_15
 TypeDefinition             : Objects
TrackUID #1                 : ATU_00001001
 ID                         : ATU_00001001
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 1
 Pack format #s             : 1
 Track format #s            : 1
TrackUID #2                 : ATU_00001002
 ID                         : ATU_00001002
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 2
 Pack format #s             : 2
 Track format #s            : 2
TrackUID #3                 : ATU_00001003
 ID                         : ATU_00001003
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 3
 Pack format #s             : 3
 Track format #s            : 3
TrackUID #4                 : ATU_00001004
 ID                         : ATU_00001004
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 4
 Pack format #s             : 4
 Track format #s            : 4
TrackUID #5                 : ATU_00001005
 ID                         : ATU_00001005
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 5
 Pack format #s             : 5
 Track format #s            : 5
TrackUID #6                 : ATU_00001006
 ID                         : ATU_00001006
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 6
 Pack format #s             : 6
 Track format #s            : 6
TrackUID #7                 : ATU_00001007
 ID                         : ATU_00001007
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 7
 Pack format #s             : 7
 Track format #s            : 7
TrackUID #8                 : ATU_00001008
 ID                         : ATU_00001008
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 8
 Pack format #s             : 8
 Track format #s            : 8
TrackUID #9                 : ATU_00001009
 ID                         : ATU_00001009
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 9
 Pack format #s             : 9
 Track format #s            : 9
TrackUID #10                : ATU_0000100a
 ID                         : ATU_0000100a
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 10
 Pack format #s             : 10
 Track format #s            : 10
TrackUID #11                : ATU_0000100b
 ID                         : ATU_0000100b
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 11
 Pack format #s             : 11
 Track format #s            : 11
TrackUID #12                : ATU_0000100c
 ID                         : ATU_0000100c
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 12
 Pack format #s             : 12
 Track format #s            : 12
TrackUID #13                : ATU_0000100d
 ID                         : ATU_0000100d
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 13
 Pack format #s             : 13
 Track format #s            : 13
TrackUID #14                : ATU_0000100e
 ID                         : ATU_0000100e
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 14
 Pack format #s             : 14
 Track format #s            : 14
TrackUID #15                : ATU_0000100f
 ID                         : ATU_0000100f
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 15
 Pack format #s             : 15
 Track format #s            : 15
TrackUID #16                : ATU_00001010
 ID                         : ATU_00001010
 Bit depth                  : 24
 Sampling rate              : 48000
 Track index                : 16
 Pack format #s             : 16
 Track format #s            : 16
TrackFormat #1              : PCM_RoomCentricCenter
 ID                         : AT_00011001_01
 Title                      : PCM_RoomCentricCenter
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 1
TrackFormat #2              : PCM_Cavern_Obj_1
 ID                         : AT_00031001_01
 Title                      : PCM_Cavern_Obj_1
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 2
TrackFormat #3              : PCM_Cavern_Obj_2
 ID                         : AT_00031002_01
 Title                      : PCM_Cavern_Obj_2
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 3
TrackFormat #4              : PCM_Cavern_Obj_3
 ID                         : AT_00031003_01
 Title                      : PCM_Cavern_Obj_3
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 4
TrackFormat #5              : PCM_Cavern_Obj_4
 ID                         : AT_00031004_01
 Title                      : PCM_Cavern_Obj_4
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 5
TrackFormat #6              : PCM_Cavern_Obj_5
 ID                         : AT_00031005_01
 Title                      : PCM_Cavern_Obj_5
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 6
TrackFormat #7              : PCM_Cavern_Obj_6
 ID                         : AT_00031006_01
 Title                      : PCM_Cavern_Obj_6
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 7
TrackFormat #8              : PCM_Cavern_Obj_7
 ID                         : AT_00031007_01
 Title                      : PCM_Cavern_Obj_7
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 8
TrackFormat #9              : PCM_Cavern_Obj_8
 ID                         : AT_00031008_01
 Title                      : PCM_Cavern_Obj_8
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 9
TrackFormat #10             : PCM_Cavern_Obj_9
 ID                         : AT_00031009_01
 Title                      : PCM_Cavern_Obj_9
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 10
TrackFormat #11             : PCM_Cavern_Obj_10
 ID                         : AT_0003100a_01
 Title                      : PCM_Cavern_Obj_10
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 11
TrackFormat #12             : PCM_Cavern_Obj_11
 ID                         : AT_0003100b_01
 Title                      : PCM_Cavern_Obj_11
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 12
TrackFormat #13             : PCM_Cavern_Obj_12
 ID                         : AT_0003100c_01
 Title                      : PCM_Cavern_Obj_12
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 13
TrackFormat #14             : PCM_Cavern_Obj_13
 ID                         : AT_0003100d_01
 Title                      : PCM_Cavern_Obj_13
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 14
TrackFormat #15             : PCM_Cavern_Obj_14
 ID                         : AT_0003100e_01
 Title                      : PCM_Cavern_Obj_14
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 15
TrackFormat #16             : PCM_Cavern_Obj_15
 ID                         : AT_0003100f_01
 Title                      : PCM_Cavern_Obj_15
 FormatDefinition           : PCM
 LinkedTo_StreamFormat_Pos  : 16
StreamFormat #1             : PCM_RoomCentricCenter
 ID                         : AS_00011001
 Title                      : PCM_RoomCentricCenter
 Format                     : PCM
 Channel format #s          : 1
 Pack format #s             : 1
 Track format #s            : 1
StreamFormat #2             : PCM_Cavern_Obj_1
 ID                         : AS_00031001
 Title                      : PCM_Cavern_Obj_1
 Format                     : PCM
 Channel format #s          : 2
 Pack format #s             : 2
 Track format #s            : 2
StreamFormat #3             : PCM_Cavern_Obj_2
 ID                         : AS_00031002
 Title                      : PCM_Cavern_Obj_2
 Format                     : PCM
 Channel format #s          : 3
 Pack format #s             : 3
 Track format #s            : 3
StreamFormat #4             : PCM_Cavern_Obj_3
 ID                         : AS_00031003
 Title                      : PCM_Cavern_Obj_3
 Format                     : PCM
 Channel format #s          : 4
 Pack format #s             : 4
 Track format #s            : 4
StreamFormat #5             : PCM_Cavern_Obj_4
 ID                         : AS_00031004
 Title                      : PCM_Cavern_Obj_4
 Format                     : PCM
 Channel format #s          : 5
 Pack format #s             : 5
 Track format #s            : 5
StreamFormat #6             : PCM_Cavern_Obj_5
 ID                         : AS_00031005
 Title                      : PCM_Cavern_Obj_5
 Format                     : PCM
 Channel format #s          : 6
 Pack format #s             : 6
 Track format #s            : 6
StreamFormat #7             : PCM_Cavern_Obj_6
 ID                         : AS_00031006
 Title                      : PCM_Cavern_Obj_6
 Format                     : PCM
 Channel format #s          : 7
 Pack format #s             : 7
 Track format #s            : 7
StreamFormat #8             : PCM_Cavern_Obj_7
 ID                         : AS_00031007
 Title                      : PCM_Cavern_Obj_7
 Format                     : PCM
 Channel format #s          : 8
 Pack format #s             : 8
 Track format #s            : 8
StreamFormat #9             : PCM_Cavern_Obj_8
 ID                         : AS_00031008
 Title                      : PCM_Cavern_Obj_8
 Format                     : PCM
 Channel format #s          : 9
 Pack format #s             : 9
 Track format #s            : 9
StreamFormat #10            : PCM_Cavern_Obj_9
 ID                         : AS_00031009
 Title                      : PCM_Cavern_Obj_9
 Format                     : PCM
 Channel format #s          : 10
 Pack format #s             : 10
 Track format #s            : 10
StreamFormat #11            : PCM_Cavern_Obj_10
 ID                         : AS_0003100a
 Title                      : PCM_Cavern_Obj_10
 Format                     : PCM
 Channel format #s          : 11
 Pack format #s             : 11
 Track format #s            : 11
StreamFormat #12            : PCM_Cavern_Obj_11
 ID                         : AS_0003100b
 Title                      : PCM_Cavern_Obj_11
 Format                     : PCM
 Channel format #s          : 12
 Pack format #s             : 12
 Track format #s            : 12
StreamFormat #13            : PCM_Cavern_Obj_12
 ID                         : AS_0003100c
 Title                      : PCM_Cavern_Obj_12
 Format                     : PCM
 Channel format #s          : 13
 Pack format #s             : 13
 Track format #s            : 13
StreamFormat #14            : PCM_Cavern_Obj_13
 ID                         : AS_0003100d
 Title                      : PCM_Cavern_Obj_13
 Format                     : PCM
 Channel format #s          : 14
 Pack format #s             : 14
 Track format #s            : 14
StreamFormat #15            : PCM_Cavern_Obj_14
 ID                         : AS_0003100e
 Title                      : PCM_Cavern_Obj_14
 Format                     : PCM
 Channel format #s          : 15
 Pack format #s             : 15
 Track format #s            : 15
StreamFormat #16            : PCM_Cavern_Obj_15
 ID                         : AS_0003100f
 Title                      : PCM_Cavern_Obj_15
 Format                     : PCM
 Channel format #s          : 16
 Pack format #s             : 16
 Track format #s            : 16
Transport #1                : Yes
 Track index #1             : ATU_00001001
  Track UID #s              : 1
 Track index #2             : ATU_00001002
  Track UID #s              : 2
 Track index #3             : ATU_00001003
  Track UID #s              : 3
 Track index #4             : ATU_00001004
  Track UID #s              : 4
 Track index #5             : ATU_00001005
  Track UID #s              : 5
 Track index #6             : ATU_00001006
  Track UID #s              : 6
 Track index #7             : ATU_00001007
  Track UID #s              : 7
 Track index #8             : ATU_00001008
  Track UID #s              : 8
 Track index #9             : ATU_00001009
  Track UID #s              : 9
 Track index #10            : ATU_0000100a
  Track UID #s              : 10
 Track index #11            : ATU_0000100b
  Track UID #s              : 11
 Track index #12            : ATU_0000100c
  Track UID #s              : 12
 Track index #13            : ATU_0000100d
  Track UID #s              : 13
 Track index #14            : ATU_0000100e
  Track UID #s              : 14
 Track index #15            : ATU_0000100f
  Track UID #s              : 15
 Track index #16            : ATU_00001010
  Track UID #s              : 16

```

**Comment by ValZapod** [comment]
BTW, now that ADM is only decoding to RAW objects, never renders stuff, it makes sense to grey out "Render target" menu.

**Comment by VoidXH** [comment]
Correct, I'll do it.

**Comment by VoidXH** [comment]
Fixed in 8fbe51f00a96ab6d74d4da939530a6c583729ce2.
